src/plots/cartesian/axes.js

```diff
@@ -249,7 +249,16 @@ axes.handleTickDefaults = function(containerIn, containerOut, coerce, axType, op
 
             if(!tickFormat && axType !== 'date') {
                 coerce('showexponent', showAttrDflt);
-                coerce('exponentformat');
+                coerce('exponentbase');
+
+                var exponentbase = containerIn.exponentbase;
+                var exponentformat = containerIn.exponentformat;
```

Comment:
@mdtusz Another upside to using 

```js
var exponentBase = coerce('exponentbase');
```

exponentBase is guaranteed to be such `typeof exponentBase === 'number'`. So you don't have to compare the `containerIn.exponentbase` against `2` and `'2'` which are both considered *numbers* by plotly.js.